File: ZenCare.Mobile/lib/models/payment_intent_response.dart
class PaymentIntentResponse {
  PaymentIntentResponse({
    required this.purchaseId,
    required this.paymentId,
    required this.stripePaymentIntentId,
    required this.clientSecret,
    required this.publishableKey,
    required this.amount,
    required this.currency,
    required this.status,
  });

  final int purchaseId;
  final int paymentId;
  final String stripePaymentIntentId;
  final String clientSecret;
  final String publishableKey;
  final double amount;
  final String currency;
  final int status;

  factory PaymentIntentResponse.fromJson(Map<String, dynamic> json) {
    return PaymentIntentResponse(
      purchaseId: json['purchaseId'] as int? ?? 0,
      paymentId: json['paymentId'] as int? ?? 0,
      stripePaymentIntentId: json['stripePaymentIntentId'] as String? ?? '',
      clientSecret: json['clientSecret'] as String? ?? '',
      publishableKey: json['publishableKey'] as String? ?? '',
      amount: (json['amount'] as num?)?.toDouble() ?? 0,
      currency: json['currency'] as String? ?? '',
      status: json['status'] as int? ?? 0,
    );
  }
}

File: ZenCare.Model/Responses/PaymentIntentResponse.cs
using ZenCare.Model.Enums;

namespace ZenCare.Model.Responses
{
    public class PaymentIntentResponse
    {
        public int PurchaseId { get; set; }

        public int PaymentId { get; set; }

        public string StripePaymentIntentId { get; set; } = string.Empty;

        public string ClientSecret { get; set; } = string.Empty;

        public string PublishableKey { get; set; } = string.Empty;

        public decimal Amount { get; set; }

        public string Currency { get; set; } = string.Empty;

        public PaymentStatus Status { get; set; }
    }
}
